lib/seadsa/DsaLocal.cc

```diff
@@ -1583,25 +1598,69 @@ void IntraBlockBuilder::visitMemTransferInst(MemTransferInst &I) {
   assert(m_graph.hasCell(*I.getDest()));
   assert(m_graph.hasCell(*I.getSource()));
 
-  // unify the two cells because potentially all bytes of source
-  // are copied into dest
   seadsa::Cell sourceCell = valueCell(*I.getSource());
   seadsa::Cell destCell = m_graph.mkCell(*I.getDest(), seadsa::Cell());
+  LOG("dsa-memtransfer", errs() << "Visiting: " << I << "\n";
+      errs() << "Before\n"; errs() << "Source Cell: " << sourceCell << "\n";
+      errs() << "Dest Cell: " << destCell << "\n";);
+
+  if (DelayMemTransfer && !m_delay_mem_transfer) {
```

Comment:
I see the need for two. One sets the mode -- we are in lazy mode, the other sets the current pass. Better names would have helped. Use AI to suggest better names.